### PR TITLE
String: return string value in the same encoding

### PR DESCRIPTION
**Pull request by andrykonchin**
Fixed String methods:

- `#chomp`
- `#succ`
- `#[]`
- `#byteslice`
- `#slice`/`#slice!`
- `#sub`
- `#split`
- `#ljust`
- `#strip!`/`#lstrip`/`#lstrip!`/`#rstrip`/`#rstrip!`
- `#downcase`
- `#upto`
- `#reverse`
- `#delete_prefix`
- `#delete_suffix`
- `#partition`

These methods might return String value in UTF-8 and ignore encoding of the original String, as far as don't set encoding explicitly and UTF-8 is used as a default one.

### Example to illustrate the issue

```ruby
"hello".encode("US-ASCII").sub("ll", "*").encoding
# => UTF-8
```

But expected one is US-ASCII.

### Tests

Not sure whether RubySpec project imports spec changes made in Natalie so I will add specs to reproduce the found issues directly in RubySpec.

### CODE REVIEW

File: src/string_object.cpp [hunk 1]
```diff
@@ -1893,7 +1893,7 @@ Value StringObject::ljust(Env *env, Value length_obj, Value pad_obj) {
     }
 
     if (padstr->string().is_empty())
-        env->raise("ArgumentError", "can't pad with an empty string");
+        env->raise("ArgumentError", "zero width padding");
```
Comment (andrykonchin) [inline src/string_object.cpp:1896]: Not relevant change, just fixed an error message (this message isn't checked in RubySpec, heh):

```ruby
# ruby 3.1.2p20

"a".ljust(10, "")
(irb):184:in `ljust': zero width padding (ArgumentError)
```

**Review by seven1m — APPROVED** [review]

**Comment by seven1m** [comment]
The his is great! This ok to merge now, or should I wait for spec coverage? (I'm ok either way.)

**Comment by andrykonchin** [comment]
Let's have it merged now if you don't mind.

**Comment by andrykonchin** [comment]
I've finally prepared specs for these changes - https://github.com/ruby/spec/pull/974. Probably we can import them after merging the PR.

**Pull request merged**